src/core/seat/keyboard.cpp

```diff
@@ -294,6 +318,33 @@ bool input_manager::handle_keyboard_key(uint32_t key, uint32_t state)
     std::vector<std::function<bool()>> callbacks;
     auto kbd = wlr_seat_get_keyboard(seat);
 
+    uint32_t leds = 0;
+    for (uint32_t i = 0; i < WLR_LED_COUNT; i++)
+    {
+        if (xkb_state_led_index_is_active(
+            kbd->xkb_state,
+            kbd->led_indexes[i]))
+        {
+            leds |= (1 << i);
+        }
+    }
+
+    if (leds & WLR_LED_NUM_LOCK)
+    {
+        wf::get_core_impl().input->locked_mods |= WF_KB_NUM;
```

Comment:
A small suggestion, 
```cpp
uint32_t& locked_mods = wf::get_core_impl().input->locked_mods;
```

Then you dont' have to type the whole thing each type (don't change if you're too lazy ;)

And please put this in a separate method.